src/ImageSharp/IO/BufferedReadStream.cs

```diff
@@ -140,7 +141,7 @@ public override int ReadByte()
 
         // Our buffer has been read.
         // We need to refill and start again.
-        if (this.readBufferIndex > this.maxBufferIndex)
+        if (this.readBufferIndex < 0 || this.readBufferIndex > this.maxBufferIndex)
```

Comment:
```suggestion
        if ((uint)this.readBufferIndex > (uint)this.maxBufferIndex)
```
Saves one comparison, as with the uint-cast negative value will be re-interpreted as values > int.MaxValue.

I know this is for streams, so the cost of reading from the stream will be dominant, but I can't stop squeezing out perf 😉